tsconfig.json

```diff
@@ -87,7 +87,8 @@
   },
   "exclude": [
     "src/devtools/client/debugger/src/workers/parser/tests/fixtures/scopes/*",
-    "node_modules"
+    "node_modules",
+    "packages/e2e-tests",
```

Comment:
This seems like a bad change. We'll lose a lot of type coverage if we exclude this whole package. Why?

Reply:
I see you added a nested TS config but it's not clear to me why this is needed.

Reply:
@markerikson - any ideas on how to manage this ts config better? the problem is that both jest and cypress add globals (e.g. for `expect()`) and they're conflicting. I was excluding the e2e tests to suppress the `cypress` import from the root devtools typechecking but the side effect seems to be that no typechecking is happening in that package at all now. I've tried a few other configs but nothing is achieving the desired result.

Reply:
@markerikson - hoping I can borrow some time from you tomorrow to get to the bottom of this

Reply:
I've minimized this change to exclude the cypress-related files so the conflict is avoided. The side effect at the moment is that the cypress config and record-cypress files are not typechecked by `tsc` though they are in VS Code.

I'm going to move ahead with this because we retain the existing typechecking of everything else with this config and the impact of those files being missed is minimal.